### PR TITLE
Configure & add Husky + lint-staged pre-commit configs

### LINKED ISSUE #40: Add pre-commit hooks for better code quality control
### Feature details?

[Git hooks](https://git-scm.com/docs/githooks) using frameworks like [husky](https://typicode.github.io/husky/#/) + [lint-staged](https://github.com/okonet/lint-staged) or [pre-commit](https://pre-commit.com/) will ensure bad coding standards & practices aren't introduced to the source code. They'll cleaned & taken care of right at the source.

Further quality control can be ensured by implementing an automated CI/CD pipeline but for starters, Git hooks is the way to go.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by tarunkishore2303**
-  Pre-commit hooks for husky has been added.
- ESLint with Google's style guide and Prettier Configuration has been added .

Closes #40

### CODE REVIEW

**Comment by Jarmos-san** [comment]
I took a quick look at the contents of the PR for now but will review it more thoroughly later tonight. Couple of things I noticed off-the-bat:

1. Contents of the PR aren't exactly on topic.
2. It might introduce potential merge conflicts down the line in future (which should be avoided if its possible).
3. Redundant configurations (which shouldn't exist in this PR in the first place).

I'll perform a more thorough review later today & will point out the stuff that needs to be cleaned up. But for starters, if you would like to take it up, see #27 if you want to make a PR about creating ESLint & Prettier configurations.

**Comment by tarunkishore2303** [comment]
I guess I've added ESLint & Prettier too with this commit

**Comment by Jarmos-san** [comment]
> I guess I've added ESLint & Prettier too with this commit

Please refer to this article - "[How to make a perfect PR](https://betterprogramming.pub/how-to-make-a-perfect-pull-request-3578fb4c112)" & clean up the unnecessary contents of this PR.

**Comment by tarunkishore2303** [comment]
Sorry for the inconvenience caused bud. Will try to clean it up !

**Review by tarunkishore2303 — COMMENTED** [review]
I think I have did my best to fine tune it. There might be lot of changes due to the formatting by prettier.  I am just trying to learn, please teach me if I am wrong.

**Comment by Jarmos-san** [comment]
Hey @tarunkishore2303 no worries, everyone's got to start somewhere & no one's perfect!

That said, I'll be reviewing the PR right now & leave out comments on sections where the files (and/or source code) isn't necessary. But to give you a gist of how to write good PRs, here's what you should know:

1. PRs should be on-topic, usually referencing a particular issue.
2. They should be as small as possible so that we maintainers can have an easier time to review it.
3. Discuss & talk about what you're trying to do before sending a PR, so that you can at least get an approval.

Keeping the aforementioned points in mind, here's what you did wrong with your PR:

1. **It's not on-topic**: Why did you add Prettier, TypeScript configs, etc to the PR when you were only supposed to add [`githooks`](https://git-scm.com/docs/githooks)?
2. Why did you add unnecessary whitespace characters to some of the files?
3. Why did you make changes to the source code in the files under the `src` directory?

File: .eslintrc.js
Comment (Jarmos-san) [inline .eslintrc.js:1]: Remove this file since it's unrelated to this PR.

File: .github/CONTRIBUTING.md
Comment (Jarmos-san) [inline .github/CONTRIBUTING.md:21]: Remove this file since it's unrelated to this PR. Also you weren't supposed to change the contents of this file without opening an issue thread.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md
Comment (Jarmos-san) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:3]: Remove this file since it's unrelated to this PR.

File: .github/dependabot.yml
Comment (Jarmos-san) [inline .github/dependabot.yml:1]: Remove this file since it's unrelated to this PR.

File: .gitignore
Comment (Jarmos-san) [inline .gitignore:6]: Why ignore `package-lock.json`? :thinking:

Also remove this file since it's unrelated to this PR.

File: .husky/pre-commit
Comment (Jarmos-san) [inline .husky/pre-commit:1]: While it's acceptable, what we need is, a combination of [lint-staged](https://github.com/okonet/lint-staged) + [husky](https://typicode.github.io/husky/#/) for performance & efficiency concerns. Give this article - "[How to Set Up Git hooks with husky and lint-staged](https://www.joshuacolvin.net/git-hooks-with-husky-and-lint-staged/)" a thorough read to understand why we need the aforementioned combination.

That said, we'll get back to cleaning it up once you've taken care of the other spam as well.

File: .prettierrc
Comment (Jarmos-san) [inline .prettierrc:1]: Spam, remove this file, make a separate PR, see #27 for more info.

File: README.md
Comment (Jarmos-san) [inline README.md:15]: Spam, remove the changes you made here.

File: package.json
Comment (Jarmos-san) [inline package.json:2]: Remove unnecessary spam from this file & then we'll get back to setting up pre-commit scripts.

File: src/pages/_app.tsx
Comment (Jarmos-san) [inline src/pages/_app.tsx:1]: Spam, remove changes you made here.

File: src/pages/index.tsx
Comment (Jarmos-san) [inline src/pages/index.tsx:4]: Spam, remove changes you made here.

File: tsconfig.json
Comment (Jarmos-san) [inline tsconfig.json:2]: Spam, remove changes you made here.

**Review by Jarmos-san — CHANGES_REQUESTED** [review]
As you can see the PR contains a lot of spam which needs to be cleaned up before we can proceed further with it.

**Please keep PRs on-topic** & make changes only to those files which you absolutely need to.

**Pull request closed without merging**
